dbms/src/Common/UInt128.h

```diff
@@ -155,10 +176,26 @@ struct UInt256HashCRC32
 };
 
 #endif
+}
 
+/// Overload hash for type casting
+namespace std
+{
+template <> struct hash<DB::UInt128>
+{
+    size_t operator()(const DB::UInt128 & u) const
+    {
+        return std::hash<DB::UInt64>()(u.first) ^ std::hash<DB::UInt64>()(u.second);
```

Comment:
Because it is broken when first == second.

The right thing is at least something like
`hash(hash(first) ^ second)`,

but it is also not suitable, because `std::hash` is trivial for integers,
and you should use something more viable like `CityHash_v1_0_2::Hash128to64`.